azurelinuxagent/ga/exthandlers.py

```diff
@@ -1207,8 +1210,9 @@ def launch_command(self, cmd, timeout=300, extension_error_code=ExtensionErrorCo
                     env = {}
                 env.update(os.environ)
                 # Always add Extension Path and version to the current launch_command (Ask from publishers)
-                env.update({EXTENSION_PATH: self.get_base_dir(),
-                            EXTENSION_VERSION: self.ext_handler.properties.version})
+                env.update({ExtCommandEnvVariable.ExtensionPath: base_dir,
```

Comment:
Does Windows pass the sequence number to all commands? We create the settings file after install so it is meaningless in the install command

Reply:
Windows only sets it for Install and Enable. I didn't see the point of not passing it to the other states too (Disable/Uninstall/Update). 

We create the settings file right before install - 
```python
        if handler_state == ExtHandlerState.NotInstalled:
            ext_handler_i.set_handler_state(ExtHandlerState.NotInstalled)
            ext_handler_i.download()
            ext_handler_i.initialize()
            **ext_handler_i.update_settings()**
            if old_ext_handler_i is None:
                ***ext_handler_i.install()***
            elif ext_handler_i.version_ne(old_ext_handler_i):
                uninstall_failed = ExtHandlersHandler._update_extension_handler_and_return_if_failed(
                    old_ext_handler_i, ext_handler_i)
        else:
            ext_handler_i.update_settings()

        ext_handler_i.enable(uninstall_failed=uninstall_failed)
```

Reply:
does windows also create the settings before install? when i wrote my extension settings were created only before enable. I am glad this has changed. I'd say it's ok to define for disable, etc.

Reply:
You are correct, windows does it after install and before enable only

```c#
// Handler installation is over, now provide the plugin with runtime settings.
                    ProcessRuntimeSettings(data);
```

Reply:
but it passes the number to install, eh? oh, my! 

i'm ok with the behavior you defined for linux